command/server/server.go

```diff
@@ -171,6 +181,26 @@ func (s *Server) setupGRPCServer(addr string) error {
 	return nil
 }
 
+func (s *Server) StartPrometheus(address string, enabled bool) error {
+	// Hook go-metrics into expvar on any /debug/metrics request, load all vars
+	// from the registry into expvar, and execute regular expvar handler.
+	if enabled {
+		exp.Exp(metrics.DefaultRegistry)
+	}
+
+	http.Handle("/memsize/", http.StripPrefix("/memsize", &Memsize))
+
+	log.Info("Starting pprof server", "addr", fmt.Sprintf("http://%s/debug/pprof", address))
+
+	go func() {
+		if err := http.ListenAndServe(address, nil); err != nil {
```

Comment:
Here you are using the embed HTTP server in Go. Please create a new instance of an HTTP server and use that one instead. Example https://stackoverflow.com/questions/25583028/how-to-write-a-simple-custom-http-server-in-go.